src/javascript-wrapper.js

```diff
@@ -17,45 +16,71 @@ var util = require('util');
 
-    // Default to no updater parameters; if no arguments, we won't go through the map
     var updaterParameters = [];
+    if (_.isEmpty(updaterArgs)) { 
 
-    // Have one or more updater parameters - get them in the right order and pass them 
-    updaterParameters = _.map( updaterArgs, function( arg ) { 
-       var state = vni.inputStates( arg );
-       if ( _.isUndefined( state ) ) { 
-           return undefined;
-       } else if (_.isArray( state )) { 
-           // Have a parameter associated with a port with multiple inputs ->
-           //  get an array of the data elements
-           return _.pluck(state, 'data');
-       } else if ( _.isObject( state ) ) {
-           // Just one input on this port parameter - get it
-           return state.data;
-       } else {
-           throw Error( "FRunupdater found an unexpected state: ", state );
-       }
-    });
+        // No updater arguments specified 
+        // TODO: handle multiple ports even though no input args
+        var states = vni.inputStates();
+        if (states && states.input && states.input.data) { 
+            updaterParameters[0] = states.input.data;
+        }
+
+    } else { 
+        // Have one or more updater parameters - get them in the right order and pass them 
+        updaterParameters = _.map(updaterArgs, function(arg) { 
+           var state = vni.inputStates(arg);
+           if (_.isUndefined(state)) { 
+               return undefined;
+           } else if (_.isArray(state)) { 
+               // Have a parameter associated with a port with multiple inputs ->
+               //  get an array of the data elements
+               return _.pluck(state, 'data');
+           } else if (_.isObject(state)) {
+               // Just one input on this port parameter - get it
+               return state.data;
+           } else {
+               throw Error("FRunupdater found an unexpected state: ", state);
+           }
+        });
+    }
     
-    // Execute the updater on the VNI context, passing the updater Parameters as the API arguments
-    return new Promise( function(resolve ) { 
+    // Get the last LM and clear error state before we call updater
+    var outputState = vni.outputState();
+    var oldOutputStateLm = outputState.lm;
+    outputState.error = undefined;
+    vni.outputState(outputState);
 
-         var results = updater.apply( vni, updaterParameters );
-         if ( results !== null ) { 
-             // If the updater returned anything, set the output state with it
-             vni.outputState( createState( vni.vnid, results ) );
+    // Execute the updater on the VNI context, passing the updater Parameters as the API arguments
+    return new Promise(function(resolve) { 
+
+         var results = updater.apply(vni, updaterParameters);
+
+         if (! _.isUndefined(results)) { 
+             // Got some results back from updater
+             Promise.resolve(results).then(function(results) { 
```

Comment:
This is forking the promise and this branch may not be executed before the framework-ondata checks the output state. Instead you should not fork the promise and include this as part of the primary promise flow.